File: csharp/ex055.cs
// Vamos melhorar o jogo que fizemos no exercício 32. A partir de
// agora, o computador vai sortear um número entre 1 e 10 e o jogador vai ter 4
// tentativas para tentar acertar.

class Ex055
{
  public int player;
  public int machine;

  public int tries = 4;

  Random randNum = new Random();
  public void GuessTheNumber()
  {
    Console.WriteLine("-----------------------");
    Console.WriteLine("Bem Vindo ao AdvinheONumero!!!");
    Console.WriteLine("-----------------------");

    while (tries == 0)
    {
      Console.WriteLine("Digite um numero de 1 a 10: ");
      player = Convert.ToInt32(Console.ReadLine());

      machine = randNum.Next(1,10);

    if (player == 0)
    {
        Console.WriteLine("Encerrando o jogo");
        break;
    }

    if (tries == 0)
    {
        Console.WriteLine("Acabaram suas tentativas");
    }

    Console.Clear();
    switch (machine)
    {
      case 1: // Tesoura
        {
            if (player == 1)
            {
              Console.WriteLine("Parabéns voce acertou!!");
            }
            else
            {
              Console.WriteLine("Infelizmente  você errou!!");
              Console.WriteLine($"Voce ainda tem {tries} tentativas");
              tries -= 1;
          }
        }
          break;
      case 2: // Papel
      {
            if (player == 2)
            {
              Console.WriteLine("Parabéns voce acertou!!");
            }
            else
            {
              Console.WriteLine("Infelizmente  você errou!!");
              Console.WriteLine($"Voce ainda tem {tries} tentativas");
              tries -= 1;
          }
        }
        break;
      case 3: //Pedra
        {
            if (player == 3)
            {
              Console.WriteLine("Parabéns voce acertou!!");
            }
            else
            {
              Console.WriteLine("Infelizmente  você errou!!");
              Console.WriteLine($"Voce ainda tem {tries} tentativas");
              tries -= 1;
          }
      }
        break;
      case 4: //Pedra
        {
            if (player == 4)
            {
              Console.WriteLine("Parabéns voce acertou!!");
            }
            else
            {
              Console.WriteLine("Infelizmente  você errou!!");
              Console.WriteLine($"Voce ainda tem {tries} tentativas");
              tries -= 1;
          }
      }
        break;
      case 5: //Pedra
        {
            if (player == 5)
            {
              Console.WriteLine("Parabéns voce acertou!!");
            }
            else
            {
              Console.WriteLine("Infelizmente  você errou!!");
              Console.WriteLine($"Voce ainda tem {tries} tentativas");
              tries -= 1;
          }
      }
        break;
      default:Console.WriteLine("Opção Invalida"); break;
    }
    }
    

  }
}

File: typescript/src/ex055.ts
// [DESAFIO] Vamos melhorar o jogo que fizemos no exercício 32. A partir de
// agora, o computador vai sortear um número entre 1 e 10 e o jogador vai ter 4
// tentativas para tentar acertar.


var readline = require('readline-sync')

let tries : number = 4;


while (tries != 0) {
  let playerOptNum2: number = parseInt(
    readline.question("Digite um numero entre 1 e 5: ")
  );

  let machineOptNum2: number = Math.floor(Math.random() * (11 - 1) + 1);

  switch (playerOptNum2) {
    case 1:
      if (machineOptNum2 === playerOptNum2) {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você acertou!!");
      } else {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você errou!!");
        if (tries != 0) {
          tries -= 1;
          console.log(`Você tem mais ${tries} tentativas`);  
        }else{
           console.log("Acabaram as tentativas");
        }
        
      }
      break;
    case 2:
      if (machineOptNum2 === playerOptNum2) {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você acertou!!");
      } else {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você errou!!");
        if (tries != 0) {
          tries -= 1;
          console.log(`Você tem mais ${tries} tentativas`);  
        }else{
           console.log("Acabaram as tentativas");
        }
      }
      break;
    case 3:
      if (machineOptNum2 === playerOptNum2) {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você acertou!!");
      } else {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você errou!!");
        if (tries != 0) {
          tries -= 1;
          console.log(`Você tem mais ${tries} tentativas`);  
        }else{
           console.log("Acabaram as tentativas");
        }
      }
      break;
    case 4:
      if (machineOptNum2 === playerOptNum2) {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você acertou!!");
      } else {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você errou!!");
        if (tries != 0) {
          tries -= 1;
          console.log(`Você tem mais ${tries} tentativas`);  
        }else{
           console.log("Acabaram as tentativas");
        }
      }
      break;

    case 5:
      if (machineOptNum2 === playerOptNum2) {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você acertou!!");
      } else {
        console.log(`voce digitou ${playerOptNum2}`);
        console.log(`o numero sorteado foi ${machineOptNum2}`);
        console.log("Você errou!!");
        if (tries != 0) {
          tries -= 1;
          console.log(`Você tem mais ${tries} tentativas`);  
        }else{
           console.log("Acabaram as tentativas");
        }
      }
      break;

    default:
      console.log(`opção invalida`);
      break;
  }
}

